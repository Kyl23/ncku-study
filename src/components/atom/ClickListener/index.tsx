import { NextPage } from 'next';
import { PropsWithChildren } from 'react';
import { EventProps, MouseEventHandle } from '../propsType';

const ClickListener: NextPage<PropsWithChildren<EventProps>> = ({
  children,
  ...eventProps
}) => {
  const onClick = eventProps.onClick as MouseEventHandle;
  return (
    <button type="button" onClick={onClick}>
      {children}
    </button>
  );
};

export default ClickListener;
