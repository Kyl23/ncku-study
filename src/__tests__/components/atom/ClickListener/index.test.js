import { fireEvent, render, screen } from '@testing-library/react';
import ClickListener from '~/src/components/atom/ClickListener';

describe('test for atom - ClickListener', () => {
    it('render Badge without props must be same', () => {
        const { container } = render(
            <ClickListener>
                <div>test</div>
            </ClickListener>
        );

        expect(container).toMatchSnapshot();
    });

    it('render Badge with event must be work correctly and must be same with non-props', () => {
        const mockClick = jest.fn();

        const { container } = render(
            <ClickListener onClick={mockClick}>
                <div>test</div>
            </ClickListener>
        );

        mockClick.mockClear();
        fireEvent.click(screen.getByRole('button'));
        expect(mockClick).toBeCalledTimes(1);
        expect(container).toMatchSnapshot();
    });
});
