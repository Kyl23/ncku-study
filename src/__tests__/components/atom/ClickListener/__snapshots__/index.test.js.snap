// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for atom - ClickListener render Badge with event must be work correctly and must be same with non-props 1`] = `
<div>
  <button
    type="button"
  >
    <div>
      test
    </div>
  </button>
</div>
`;

exports[`test for atom - ClickListener render Badge without props must be same 1`] = `
<div>
  <button
    type="button"
  >
    <div>
      test
    </div>
  </button>
</div>
`;
